### ISSUE TITLE
Configure Rest Assured for https

### ISSUE THREAD

**Issue by Kin-Z** [quarkusio/quarkus#5877]
**Description**
It would be nice if could easily configure Rest Assured such that it uses https/ssl only.

**Implementation ideas**
Maybe an additional configuration property could be added.

**Comment by lordofthejars** [comment]
@geoand it was created but no feedback

**Comment by geoand** [comment]
@lordofthejars Thanks.

Under what condition do you envision that we would revert to https? I mean how would this be configured by the user?

**Comment by lordofthejars** [comment]
Well, there is a lot of options and I don't know what is best. 

* Adding a boolean flag on the annotation to indicate if we want to use the HTTPS or the HTTP port. Default to HTTP. 
* As a configuration property in `application.properties`. I don't like this approach as the configuration of the test is outside the test.
* Inject the port in property and configure RestAssured manually.
* Create a specific annotation like `@SecureTest` or something like this that indicates that the test is using the HTTPS port.

**Comment by geoand** [comment]
I was also thinking of 1) or 4), but I'm not sure either...

@gsmet @gastaldi what do you think?

**Comment by gastaldi** [comment]
There is more to consider when hitting an HTTPS URL than just the port like, for example, the truststore to be used or certificates to be used for authentication but I think the majority of the use cases will use `given().relaxedHTTPSValidation()` anyway

**Comment by gastaldi** [comment]
Perhaps the URL injected by `@TestHTTPResource` could have a secure attibute?

**Comment by mkouba** [comment]
Although I like the annotation approach I think that we should be consistent and allow the users to configure the "secure connection" through config (system property, properties file, etc.) like we do for `quarkus.http.test-port`, `quarkus.http.host` and `quarkus.http.root-path`. For example we could introduce a "quarkus.secure-connection" config property read the value in `RestAssuredURLManager.setURL()`.

**Comment by lordofthejars** [comment]
I agree with you, but what I see about this approach is that as a tester viewer you never know directly if you are testing http or https until you look at `application.properties`. That's fine, maybe it is not important at all in this case, but usually, for me as a test viewer I really like to have as much as information in the test itself as possible.

**Comment by mkouba** [comment]
You don't see the host, port, etc. anyway. These are just defaults and you can always set the values manually.

>  ...until you look at application.properties

Actually, I think that you can use a system property or ENV in which case you don't see it anywhere ;-).

**Comment by geoand** [comment]
> for me as a test viewer I really like to have as much as information in the test itself as possible.

+1

**Comment by gastaldi** [comment]
This feature has been implemented in the current codebase. Here's a summary of what's available:

### `@TestHTTPResource(tls = true)`

The `@TestHTTPResource` annotation has a `tls` attribute that switches to HTTPS URLs using `quarkus.http.test-ssl-port` (default 8444):

```java
@TestHTTPResource(value = "/endpoint", tls = true)
URL httpsUrl;
```

### Automatic HTTPS when insecure requests are disabled

REST Assured automatically switches to HTTPS when `quarkus.http.insecure-requests` is not `"enabled"` (handled in `QuarkusTestExtension`).

### Global config flag

`quarkus.http.test-ssl-enabled=true` forces all `@TestHTTPResource` injections to use HTTPS URLs.

### MTLS support

For mutual TLS with client certificates, combine `@TestHTTPResource(tls = true)` with REST Assured's `RequestSpecBuilder`:

```java
@TestHTTPResource(tls = true)
URL url;

RequestSpecification spec = new RequestSpecBuilder()
    .setBaseUri(String.format("%s://%s", url.getProtocol(), url.getHost()))
    .setPort(url.getPort())
    .setKeyStore("client-keystore.jks", "password")
    .setTrustStore("client-truststore.jks", "password")
    .build();

given().spec(spec).get("/path").then().statusCode(200);
```

I believe this issue can be closed.

**Comment by gastaldi** [comment]
Closing as the requested functionality is already available.
